pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -308,24 +308,50 @@ type VitessShardStatus struct {
 	// HasMaster is a condition indicating whether the Vitess topology
 	// reflects a master for this shard.
 	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
+
 	// HasInitialBackup is a condition indicating whether the initial backup
 	// has been seeded for the shard.
 	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
+
 	// MasterAlias is the tablet alias of the master according to the global
 	// shard record. This could be empty either because there is no master,
 	// or because the shard record could not be read. Check the HasMaster
 	// condition whenever the distinction is important.
 	MasterAlias string `json:"masterAlias,omitempty"`
 
+	// BackupLocations reports information about the backups for this shard in
+	// each backup location.
+	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+
 	// Idle is a condition indicating whether the shard can be turned down.
 	// If Idle is True, the shard is not part of the active shard set
 	// (partitioning) for any tablet type in any cell, so it should be safe
 	// to turn down the shard.
 	Idle corev1.ConditionStatus `json:"idle,omitempty"`
 
-	// BackupLocations reports information about the backups for this shard in
-	// each backup location.
-	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	Conditions map[VitessShardConditionType]*VitessShardCondition `json:"conditions,omitempty"`
+}
+
+// VitessShardConditionType is a valid value for VitessShardCondition.Type
+type VitessShardConditionType string
+
+// VitessShardCondition contains details for the current condition of this VitessShard.
+type VitessShardCondition struct {
+	// Type is the type of the condition.
+	Type VitessShardConditionType `json:"type"`
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
```

Comment:
Perhaps we should enforce this with an enum validation, especially since we're opening this up for dynamic use by other controllers. If any other values sneak in, it could cause difficult-to-diagnose bugs.

Reply:
I think that's a good idea. I'll add Kubebuilder enum validation.

Reply:
Done https://github.com/planetscale/vitess-operator/pull/31/commits/5011b4b6b77b9d5a6a9d82a45cc04ce04ff3e14c.